homeassistant/components/bmw_connected_drive/config_flow.py

```diff
@@ -94,14 +94,9 @@ def async_get_options_flow(
         return BMWOptionsFlow(config_entry)
 
 
-class BMWOptionsFlow(config_entries.OptionsFlow):
+class BMWOptionsFlow(config_entries.OptionsFlowWithConfigEntry):
     """Handle a option flow for MyBMW."""
 
-    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
-        """Initialize MyBMW option flow."""
-        self.config_entry = config_entry
-        self.options = dict(config_entry.options)
-
```

Comment:
We cannot convert to `SchemaCommonFlowHandler` because `validate_user_input` does not currently support returning a coroutine. Should we adjust to enable async calls in `validate_user_input`?

Reply:
Yeah, I think that makes sense.
Should we change so both coroutine functions and regular functions are allowed, or change so only coroutine functions are allowed?

Reply:
Happy to simply use `OptionsFlowWithConfigEntry` for now, and revisit the async if it comes up in other integrations.

Reply:
I think it's safer at this stage to simply move to `OptionsFlowWithConfigEntry` and not make changes to the custom reload handling. It seems to be here to avoid a potential race with the coordinator:
https://github.com/home-assistant/core/blob/79b400d71b1a9ddfeda4eb388c35a2de51d8f55d/homeassistant/components/bmw_connected_drive/coordinator.py#L83